kythe/go/localrun/localrun.go

```diff
@@ -131,57 +150,59 @@ func (l Language) indexerPath() string {
 	return l.metadata().indexerPath
 }
 
-// Valid determines of the provided language is a valid enum value.
-func (l Language) Valid() bool {
-	return l <= TypeScript
-}
-
 func (l Language) metadata() metadata {
 	return languageMetadataMap[l]
 }
 
-func AllLanguages() LanguageList {
-	languages := LanguageList{}
+// AllLanguages returns a language set that contains all available languages.
+func AllLanguages() LanguageSet {
+	languages := LanguageSet{}
 	for c := Language(0); c.Valid(); c++ {
-		languages = append(languages, c)
+		languages.Set(c)
 	}
 	return languages
 }
 
-type LanguageList []Language
+// LanguageSet is a set implementation that tracks langages.
+type LanguageSet map[Language]struct{}
```

Comment:
nit: we normally use `bitbucket.org/creachadair/stringset` for sets of strings

Reply:
Unfortunately this isn't a set of strings, it is a set of languages. Looking at stringset it doesn't look like it can work on `interface{}` types so we have to write our own ☹️. Maybe we will get generics some day.

Reply:
The language enum could be backed by a `string` representation instead of an `int` to simplify this.